diff --git a/dist/node/chunks/config.js b/dist/node/chunks/config.js
index 2b39a20c497ee5c9f1cefdf1781df9955b50b97d..33cf07bd28a5ff00e97bc286d81ffdd2db29795b 100644
--- a/dist/node/chunks/config.js
+++ b/dist/node/chunks/config.js
@@ -28714,6 +28714,7 @@ var EnvironmentPluginContainer = class {
 			}
 			const pluginResolveStart = debugPluginResolve ? performance$1.now() : 0;
 			const handler = getHookHandler(plugin.resolveId);
+			if (!handler) continue;
 			const result = await this.handleHookPromise(handler.call(ctx, rawId, importer, normalizedOptions));
 			if (!result) continue;
 			if (typeof result === "string") id = result;
@@ -28756,6 +28757,7 @@ var EnvironmentPluginContainer = class {
 				}
 			});
 			const handler = getHookHandler(plugin.load);
+			if (!handler) continue;
 			const result = await this.handleHookPromise(handler.call(ctx, id, options$1));
 			if (result != null) {
 				if (isObject(result)) ctx._updateModuleInfo(id, result);
@@ -28793,6 +28795,7 @@ var EnvironmentPluginContainer = class {
 			const start = debugPluginTransform ? performance$1.now() : 0;
 			let result;
 			const handler = getHookHandler(plugin.transform);
+			if (!handler) continue;
 			try {
 				result = await this.handleHookPromise(handler.call(ctx, code, id, optionsWithSSR));
 			} catch (e$1) {
