diff --git a/dist/node/chunks/config.js b/dist/node/chunks/config.js
index 9a45043275249a25101fd7991ffcfc316478840e..1ad296340b8cf4e170dd4c0f59fe91b4643e3559 100644
--- a/dist/node/chunks/config.js
+++ b/dist/node/chunks/config.js
@@ -28794,6 +28794,7 @@ var EnvironmentPluginContainer = class {
 			}
 			const pluginResolveStart = debugPluginResolve ? performance$1.now() : 0;
 			const handler = getHookHandler(plugin.resolveId);
+			if (!handler) continue;
 			const result = await this.handleHookPromise(handler.call(ctx, rawId, importer, normalizedOptions));
 			if (!result) continue;
 			if (typeof result === "string") id = result;
@@ -28836,6 +28837,7 @@ var EnvironmentPluginContainer = class {
 				}
 			});
 			const handler = getHookHandler(plugin.load);
+			if (!handler) continue;
 			const result = await this.handleHookPromise(handler.call(ctx, id, options$1));
 			if (result != null) {
 				if (isObject(result)) ctx._updateModuleInfo(id, result);
@@ -28873,6 +28875,7 @@ var EnvironmentPluginContainer = class {
 			const start = debugPluginTransform ? performance$1.now() : 0;
 			let result;
 			const handler = getHookHandler(plugin.transform);
+			if (!handler) continue;
 			try {
 				result = await this.handleHookPromise(handler.call(ctx, code, id, optionsWithSSR));
 			} catch (e$1) {
